CONTRIBUTING.md

```diff
@@ -25,22 +25,535 @@ Below is a sample of how additional vehicle directories should be created.
 ## Getting Started
 
 * Using the repo's Issues section, create a new issue and:
-  * Clearly describe the issue including steps to reproduce when it is a bug.
-* Fork the repository on GitHub
+  * Clearly describe the issue including steps to reproduce when it is a bug
+  * Create a branch using `git fork` in the repository on GitHub
 
-## Making Changes
+## Making Contributions
+
+* Do not work directly on the `master` branch; changes to the master branch will be removed
+* Create a topic branch from where you want to base your work
+  * You'll likely `fork` from of the master branch
+  * To quickly create a topic branch based on master:
+  ```
+  git checkout -b fix/master/my_contribution master
+  ```
 
-* Create a topic branch from where you want to base your work.
-  * You'll likely branch off of the master branch.
-  * To quickly create a topic branch based on master; `git checkout -b
-    fix/master/my_contribution master`. Please avoid working directly on the
-    `master` branch.
 * Write a good commit message.
+#### Commit Messages
+* Use the present tense ("Add feature" not "Added feature")
+* Use the imperative mood ("Move resistor to..." not "Moves resistor to...")
+* Limit the first line to 72 characters or less
+* Reference issues and pull requests specifically
+* When changing documentation only, include [ci skip] in the commit description
+
+### Contributing Code
+
+* Code should conform to the coding standard (see below)
+* Push your changes to a topic branch in your branch of the repository
+* Submit a pull request to the repository in the PolySync organization
+* Update your github issue to mark that you have submitted code and are ready for it to be reviewed (Status: Ready for Merge)
+  * Include a link to the pull request in the ticket
+* The PolySync team will review all pull requests on a weekly basis
+* __Code contributed should include unit tests__ that demonstrate the code functions as expected
+
+### Contributing a Diagram
+
+* The diagrams were created using the Eagle PCB Design tool (https://cadsoft.io/#)
+    * All board diagrams should use this tool
+* Modifications should include tests to demonstrate the the modified board works as expected
+
+### Submitting Enhancement Requests
+
+Enhancement suggestions are tracked as GitHub issues:
+
+* Use a clear and descriptive title for the issue to identify the suggestion
+* Provide a step-by-step description of the suggested enhancement in as many details as possible
+* Provide specific examples to demonstrate the steps
+    * Include snippets in the examples as Markdown code blocks ( using the \`\`\` format )
+* Describe the current behavior and how this enhancement improves or changes it
+* Explain why this enhancement is useful to most users
+* Specify which version of OSCC this change applies to
+    * The versions can be found in the Releases tab on GitHub
+
+### Pull Request Labels
+* Fill in the required template
+* Follow the style guides

```

Comment:
'style guides' do we have multiple style guides, or should this be singular instead of plural?